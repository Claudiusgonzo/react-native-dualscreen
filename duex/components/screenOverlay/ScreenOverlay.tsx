import React, { Fragment} from 'react';

import {
  View, ViewStyle, StyleProp,
} from 'react-native';
import { ScreenOverlayStyles } from './ScreenOverlay.styles';

interface IScreenOverlayProps {
   /**
   * Content of the overlay
   */
    children: React.ReactElement<any>;
   /**
   * If true, the overlay is visible
   */
    isVisible:  boolean;
   /**
   * Style for the backdrop
   */
    backdropStyle?: StyleProp<ViewStyle>;
   /**
   * Style of the actual overlay
   */
    overlayStyle?: StyleProp<ViewStyle>;
   /**
   * Callback when user touches the backdrop
   */
    onBackdropPress?:()=> void;
}

const ScreenOverlay = (props: IScreenOverlayProps) => {
  return (
    <Fragment>
      {props.isVisible && (
        <View
          style={[ScreenOverlayStyles.backdrop, props.backdropStyle]}
            onTouchEndCapture={event => {
              //check if we are clicking ourselves or a different component
              if (
                String(event.currentTarget) === String(event.nativeEvent.target)
              ) {
                if(props?.onBackdropPress)
                {
                  props?.onBackdropPress!();
                }
              }
            }}>
            <View
              pointerEvents={'box-none'}
              style={[ScreenOverlayStyles.overlay, props.overlayStyle]}>
              {props.children}
            </View>
          </View>
      )}
    </Fragment>
  );
};

export default ScreenOverlay;
