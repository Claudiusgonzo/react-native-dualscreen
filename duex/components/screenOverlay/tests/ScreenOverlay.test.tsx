import React, { Fragment } from 'react';
import renderer from 'react-test-renderer';
import ScreenOverlay from '../ScreenOverlay';

describe('ScreenOverlay Tests', () => {
  it('should render', () => {
    // Arrange
    jest.useFakeTimers();

    // Act
    const tree = renderer.create(
      <ScreenOverlay isVisible={true}>
        <Fragment />
      </ScreenOverlay>
    );

    // Assert
    expect(tree).toMatchSnapshot();
  });
});
