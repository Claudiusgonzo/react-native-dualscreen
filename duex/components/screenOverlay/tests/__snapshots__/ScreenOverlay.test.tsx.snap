// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenOverlay Tests should render 1`] = `
<View
  onTouchEndCapture={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0,0,0,0.5)",
        "bottom": 0,
        "flex": 1,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      undefined,
    ]
  }
>
  <View
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "backgroundColor": "#f2f2f2",
          "zIndex": 5,
        },
        undefined,
      ]
    }
  />
</View>
`;
